server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -273,6 +301,40 @@ public static MetadataVersion latest() {
         return VALUES[VALUES.length - 1];
     }
 
+    public Optional<MetadataVersion> previous() {
+        int idx = this.ordinal();
```

Comment:
The indexing for `ordinal()` starts at `0` right? With `MetadataVersion.UNINITIALIZED` at index 0 (which I'm assuming we don't want to count as a valid previous version), should the conditional be `if (idx > 1)`?

Reply:
Good catch! I think this was left over from when I had put UNINITIALIZED into VALUES. I opted against that since it would be a gotcha for future developers to encounter a version that was not really a version.